scripts/verify-ctype-algorithm-include-order.sh

```diff
@@ -21,16 +21,37 @@ declare -gr green="\\e[32m"
 declare -gr cyan="\\e[36m"
 declare -gr reset="\\e[0m"
 
+# Path to our define script
+write_defines="./scripts/dump-preprocessor-defines.sh"
+
 # Round up defines files
-# shellcheck disable=SC2207
-defines=( $(find . -name '*.defines') )
-
-# Ensure we have the expected quantity
-actual="${#defines[@]}"
-expected="$(find . -name '*.cpp' -o -name '*.c' | wc -l)"
-if (( "$actual" != "$expected" )); then
-    echo "We should have $expected .defines files but $actual were found"
-	exit 1
+if [[ "$#" == "0" ]]; then
+	"$write_defines"
+
+	# shellcheck disable=SC2207
+	defines=( $(find . -name '*.defines') )
```

Comment:
Thankfully we don't have source files with spaces in the name ;)

Reply:
Python re-write :-) Absolutely; if/when time permits -- the sooner we don't need these the better!  Bash scripts always seem to out grow their welcome! But, alas, remain useful until a leaner counterpart is ready.

If I quote the find, then the array becomes one giant string, the hash-counter `"${#defines[@]}` only returns one instead of the correct number of files, and we have to split the string on spaces in the for loop using something like this: `for f in ${defines[*]}; do ...` (which shellcheck also complains about).  Where as if I don't quote the find, then we get correct $# hash counting and can use quotes in the loop `for f in "${defines[@]}"`.  So give the two choices, I went with the one with fewer problems. 

Yeah.. python would be much better in this case because it's just text processing!